src/config.cpp

```diff
@@ -473,7 +486,16 @@ void loadServerConfigFromString(char *config) {
         } else if ((!strcasecmp(argv[0],"slaveof") ||
                     !strcasecmp(argv[0],"replicaof")) && argc == 3) {
             slaveof_linenum = linenum;
-            replicationAddMaster(argv[1], atoi(argv[2]));
+            if (!strcasecmp(argv[1], "no") && !strcasecmp(argv[2], "one")) {
+                listRelease(g_pserver->masters);
```

Comment:
This doesn't free any of the data handing off the master info structs in the list, and leaves g_pserver->masters as a dangling pointer so we crash the next time we touch the list.

We should have a test here that exercises this case (has a replicaof, followed by replicaof no one, followed by a real replicaof).  Check with ASAN there are no leaks and dangling pointers.